File: App/src/app/domain/models/tbl-detalle-factura/tbl-detalle-factura.ts
import { CatProducto } from '../cat-producto/cat-producto';
import { TblFactura } from '../tbl-factura/tbl-factura';

export interface TblDetalleFactura {
  id: number;
  idFactura: number;
  idProducto: number;
  cantidadDeProducto: number;
  precioUnitario: number;
  subTotal: number;
  notas: string;
  createdAt: Date;
  updatedAt: Date;
  deletedAt: Date | null;
  factura: TblFactura;
  producto: CatProducto;
}

File: Api/Models/TblDetalleFactura.cs
namespace Api.Models
{
    public class TblDetalleFactura
    {
        // Primary Key
        public int Id { get; set; }
        // Properties
        public required int IdFactura { get; set; }
        public required int IdProducto { get; set; }
        public required int CantidadDeProducto { get; set; }
        public decimal PrecioUnitario { get; set; }
        public decimal SubTotal { get; set; }
        public required string? Notas { get; set; }
        // Timestamps
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }
        // Foreign Keys
        public TblFactura? Factura { get; set; }
        public CatProducto? Producto { get; set; }
    }
}